components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt

```diff
@@ -232,18 +232,8 @@ open class PlacesHistoryStorage(
         return places.importVisitsFromFennec(dbPath)
     }
 
-    /**
-     * This should be removed. See: https://github.com/mozilla/application-services/issues/1877
-     *
-     * @return raw internal handle that could be used for referencing underlying [PlacesApi]. Use it with SyncManager.
-     */
-    override fun getHandle(): Long {
-        return places.getHandle()
-    }
-
     override fun registerWithSyncManager() {
-        // See https://github.com/mozilla-mobile/android-components/issues/10128
-        throw NotImplementedError("Use getHandle instead")
+        return places.registerWithSyncManager()
```

Comment:
It would be great if we could add this in `PlacesHistoryStorageTest.kt`. Not sure how tricky that would be 🤔

Reply:
In the same vein, are there any changes that should be made to `PlacesBookmarksStorageTest.kt` as well?

Reply:
I would love love to test the sync manager integration more, but I can't think of how to do it exactly.  We could add a sync manager function that returns which engines are registered to and test that things seem to be linking up correctly.  Testing the actual sync process would be very tricky though.

Reply:
Another issue with testing is that adding methods to sync manager is tricky before it's using UniFFI. I just updated the s[ync manager UniFFI PR](https://github.com/mozilla/application-services/pull/4645) with [some tests for `registerWithSyncManager`](https://github.com/mozilla/application-services/pull/4645/commits/d4ea9bc1d550d3d3035a04e7ed8b5bca03787f69)

This is the best I can think of.  It seems like that should test that places is getting registered with the sync manager correctly at least.  It would be great to test the actual syncing functionality, but that's a lot more work.

Do you think this one is in good enough shape that we can merge the a-s branch and make an a-s release with the new code?  Then we can test this better against the actual a-s code and make the CI pass.

Reply:
Yeah, let's go ahead and get the a-s branch merged 👍